routing/http/server/server.go

```diff
@@ -369,10 +380,33 @@ func (s *server) provide(w http.ResponseWriter, httpReq *http.Request) {
 	writeJSONResult(w, "Provide", resp)
 }
 
-func (s *server) findPeersJSON(w http.ResponseWriter, peersIter iter.ResultIter[*types.PeerRecord]) {
+func (s *server) findPeersJSON(w http.ResponseWriter, peersIter iter.ResultIter[*types.PeerRecord], filterAddrs, filterProtocols []string) {
 	defer peersIter.Close()
 
-	peers, err := iter.ReadAllResults(peersIter)
+	// Convert PeerRecord to Record so that we can reuse the filtering logic from findProviders
+	mappedIter := iter.Map(peersIter, func(v iter.Result[*types.PeerRecord]) iter.Result[types.Record] {
+		if v.Err != nil || v.Val == nil {
```

Comment:
Do you mean to return the value and ignore the error when `v.Err != nil` and `v.Val` is not `nil`? If there is a use case for this, then a comment would be god here.

Reply:
The idea was to return `nil` when there's an error **or** the value is nil.